Automattic/jetpack
# Modules: Make activate() method Jetpack-the-plugin agnostic

It's an ugly refactor, stuffs Jetpack-specific logic into a class_exists check. However it allows standalone plugins to use the Modules()->activate() method. 

#### Changes proposed in this Pull Request:

- Refactors `Modules()->activate()` to execute Jetpack-specific logic only when Jetpack is present.
- Updates Search activation method to use it directly. 
- Updates Search deactivation method to use the Module class method - there was no Jetpack-specific logic there. 

#### Jetpack product discussion


- p1652392614473149-slack-CBG1CP4EN
- pdWQjU-o-p2

#### Does this pull request change what data or activity we track or use?


No 

#### Testing instructions:






- Set up a site with Search plan
- Enable/disable the Search setting from the Search plugin when Jetapck is not active. 
- When activating Jetpack, make sure the status is accurate. 

Make sure that activating the modules from Jetpack also works: 
- From the settings UI 
- From /wp-admin/admin.php?page=jetpack_modules
- From wp-cli 
- From Calypso settings
